### PR TITLE
Integration docs

### PR DESCRIPTION
**Pull request by wtrocki**
Documentation for AMQ and Keycloak integrations

### CODE REVIEW

File: docs/authentication/authentication.md [hunk 1]
```diff
@@ -1,7 +1,25 @@
...
+and prevent from unautorized access.
+
+## Out of the box Authorization on top of the Graphback
+
+Graphback recomends using following authorization libraries
+
+- [Passport.js](http://www.passportjs.org/) if you looking for in application authorization and authentication mechanisms.
+- [Keycloak](https://www.keycloak.org/) SSO - Standalone SSO server offering integration with various social OAuth logins and federating users.
+
+Graphback provides out of the box support for Keycloak SSO thanks to @graphback/keycloak-authz that utilizes
```
Comment (machi1990) [inline docs/authentication/authentication.md:24]:
```suggestion
Graphback provides out of the box support for Keycloak SSO thanks to [@graphback/keycloak-authz](https://www.npmjs.com/package/@graphback/keycloak-authz) that utilizes
```

File: docs/authentication/keycloak.md [hunk 2]
```diff
@@ -4,4 +4,238 @@ title: Authentication with Keycloak
...
 
-## TODO
\ No newline at end of file
+## Graphback Keycloak Authz
+
+Graphback Keycloak Authz enables [Keycloak](https://www.keycloak.org/) integration in [Graphback](https://graphback.dev) based applications. This enables you to declaratively add authorization capabilities like role based access on top of the CRUD model that is used within Graphback.
+
+This package is designed to work with [`keycloak-connect`](https://www.npmjs.com/package/keycloak-connect) and [`keycloak-connect-graphql`](https://www.npmjs.com/package/keycloak-connect-graphql). `keycloak-connect` is the official Keycloak middleware for Express applications. `keycloak-connect-graphql` provides deeper Keycloak integration into GraphQL servers.
+
+> NOTE: This package is an early alpha and not officially supported by Graphback
```
Comment (machi1990) [inline docs/authentication/keycloak.md:13]:
```suggestion
:::note
This package is an early alpha and not officially supported by Graphback
:::
```

File: docs/authentication/keycloak.md [hunk 3]
```diff
@@ -4,4 +4,238 @@ title: Authentication with Keycloak
...
+- Tasks `allTasksComments` field has `commenter` role applied. Fetching `Comment` object will require `commenter` role for any user field fetched
+
+:::info
+Due to limitations of the Graphback `relations` authorization works only on `OneToMany` relationships.
+`OneToOne` relationships will utilize `read` config of the relationship target.
+:::
+
+## Field Updates Authorization
+
+Developers can prevent from unautorized updates of the certain fields for `create` and `update` operations.
```
Comment (machi1990) [inline docs/authentication/keycloak.md:87]:
```suggestion
Developers can prevent from unauthorized updates of the certain fields for `create` and `update` operations.
```

File: docs/authentication/keycloak.md [hunk 4]
```diff
@@ -4,4 +4,238 @@ title: Authentication with Keycloak
...
+```
+
+With this configuration the following rules are in place.
+
+- Tasks `secret` field has `admin` role applied. `createTask` and `updateTask` will require `admin` role if
+  secret field was provided as part of the input type.
+
+:::info
+When designing your schema you need to consider impact of the authorization rules on your types.
+Any field that is required
```
Comment (machi1990) [inline docs/authentication/keycloak.md:106]: This sentence does not connect well with the rest of the note.

File: docs/authentication/keycloak.md [hunk 5]
```diff
@@ -4,4 +4,238 @@ title: Authentication with Keycloak
...
+```
+
+:::info
+Dynamic filtering is an **experimental** feature due to lack of strong typings on `profileInfo` and `filter`.
+Please review Graphback `QueryFilter` definitions and Keycloak profile information objects before building filters.
+:::
+
+## Your own Resolvers
+
+`keycloak-authz` package provides out of the box handlers for the graphback generated resolvers.
```
Comment (machi1990) [inline docs/authentication/keycloak.md:172]:
```suggestion
`keycloak-authz` package provides out of the box handlers for the Graphback generated resolvers.
```

File: docs/authentication/keycloak.md [hunk 6]
```diff
@@ -4,4 +4,238 @@ title: Authentication with Keycloak
...
+Developers can utilize authorization in their own resolvers and custom code.
+For more information please reffer to the https://github.com/aerogear/keycloak-connect-graphql library documentation
```
Comment (machi1990) [inline docs/authentication/keycloak.md:174]:
```suggestion
For more information please refer to the https://github.com/aerogear/keycloak-connect-graphql library documentation
```

File: docs/authentication/keycloak.md [hunk 7]
```diff
@@ -4,4 +4,238 @@ title: Authentication with Keycloak
...
+  Report: {
+    create: { roles: ['admin'] },
+    update: { roles: ['admin'] },
+    delete: { roles: ['admin'] },
+  },
+};
+```
+
+:::info
+`CrudServiceAuthConfig` interface defines all possible keys that each model can define.
```
Comment (machi1990) [inline docs/authentication/keycloak.md:50]: Once we have something like https://github.com/aerogear/graphback/issues/1702 we may be able to add link for the `CrudServiceAuthConfig` interface and all the other API.

**Comment by wtrocki** [comment]
Great point about API docs. I have been thinking about pointing to master but then seen your suggestions this morning so we can do that better

**Comment by machi1990** [comment]
> Great point about API docs. I have been thinking about pointing to master but then seen your suggestions this morning so we can do that better

:+1: sure.  

Can we publish the docs website? @wtrocki

**Comment by wtrocki** [comment]
Go for it

**Comment by wtrocki** [comment]
Feature wise we just got very specific requirements that are not that generic itself. 
In future we can look into something like firebase:
https://firebase.google.com/docs/firestore/solutions/role-based-access

Created follow up for those

**Pull request merged**